### ISSUE TITLE
New navbar actions menu is not accessible

### ISSUE THREAD

**Issue by Betree** [opencollective/opencollective#3696]
We should be able to open through the menu with tab + enter keys.

To achieve that:
- Use a `StyledButton` (can make it "invisible" with `buttonStyle="secondary" isBorderless` + `buttonSize`)
- Or use `tabIndex`
